pkg/gadgets/run/tracer/tracer.go

```diff
@@ -894,3 +886,7 @@ func (t *Tracer) SetEventHandlerArray(handler any) {
 	}
 	t.eventArrayCallback = nh
 }
+
+func (t *Tracer) SetSocketEnricherMap(m *ebpf.Map) {
+	t.socketEnricherMap = m
```

Comment:
Does it mean that image-based gadgets would still inconditionally create the socket enricher, even if they don't use it?

Reply:
You're right. It's still a pending issue we have with other operators as well (https://github.com/inspektor-gadget/inspektor-gadget/issues/2167), I gave it a try on https://github.com/inspektor-gadget/inspektor-gadget/pull/2168.